// @tribeplatform/gql-client/types/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

exports.AccessGroupEntityType = void 0;
(function (AccessGroupEntityType) {
    AccessGroupEntityType["NETWORK"] = "network";
    AccessGroupEntityType["POST"] = "post";
    AccessGroupEntityType["SPACE"] = "space";
})(exports.AccessGroupEntityType || (exports.AccessGroupEntityType = {}));
exports.ActionStatus = void 0;
(function (ActionStatus) {
    ActionStatus["FAILED"] = "failed";
    ActionStatus["SUCCEEDED"] = "succeeded";
})(exports.ActionStatus || (exports.ActionStatus = {}));
exports.AppCollaboratorType = void 0;
(function (AppCollaboratorType) {
    AppCollaboratorType["COLLABORATOR"] = "COLLABORATOR";
    AppCollaboratorType["OWNER"] = "OWNER";
})(exports.AppCollaboratorType || (exports.AppCollaboratorType = {}));
exports.AppInstallationStatus = void 0;
(function (AppInstallationStatus) {
    AppInstallationStatus["DELETED"] = "DELETED";
    AppInstallationStatus["DISABLED"] = "DISABLED";
    AppInstallationStatus["ENABLED"] = "ENABLED";
})(exports.AppInstallationStatus || (exports.AppInstallationStatus = {}));
exports.BadgeType = void 0;
(function (BadgeType) {
    BadgeType["ADMIN"] = "Admin";
    BadgeType["MANUAL"] = "Manual";
    BadgeType["MODERATOR"] = "Moderator";
    BadgeType["NEW_MEMBER"] = "NewMember";
})(exports.BadgeType || (exports.BadgeType = {}));
exports.CollectionListOrderByEnum = void 0;
(function (CollectionListOrderByEnum) {
    CollectionListOrderByEnum["CREATED_AT"] = "CREATED_AT";
    CollectionListOrderByEnum["CUSTOM_ORDERING_INDEX"] = "CUSTOM_ORDERING_INDEX";
    CollectionListOrderByEnum["UPDATED_AT"] = "UPDATED_AT";
})(exports.CollectionListOrderByEnum || (exports.CollectionListOrderByEnum = {}));
exports.CustomCodePosition = void 0;
(function (CustomCodePosition) {
    CustomCodePosition["BODY"] = "BODY";
    CustomCodePosition["HEAD"] = "HEAD";
})(exports.CustomCodePosition || (exports.CustomCodePosition = {}));
exports.CustomFieldPrivacyOptions = void 0;
(function (CustomFieldPrivacyOptions) {
    CustomFieldPrivacyOptions["ADMIN"] = "ADMIN";
    CustomFieldPrivacyOptions["OWN"] = "OWN";
})(exports.CustomFieldPrivacyOptions || (exports.CustomFieldPrivacyOptions = {}));
exports.CustomFieldType = void 0;
(function (CustomFieldType) {
    CustomFieldType["ARRAY"] = "array";
    CustomFieldType["BOOLEAN"] = "boolean";
    CustomFieldType["DATE"] = "date";
    CustomFieldType["NUMBER"] = "number";
    CustomFieldType["OBJECT"] = "object";
    CustomFieldType["RELATION"] = "relation";
    CustomFieldType["RICH_TEXT"] = "richText";
    CustomFieldType["TEXT"] = "text";
})(exports.CustomFieldType || (exports.CustomFieldType = {}));
exports.CustomFieldValidators = void 0;
(function (CustomFieldValidators) {
    CustomFieldValidators["ALL_OF"] = "allOf";
    CustomFieldValidators["ANY_OF"] = "anyOf";
    CustomFieldValidators["ENUM"] = "enum";
    CustomFieldValidators["EXCLUSIVE_MAXIMUM"] = "exclusiveMaximum";
    CustomFieldValidators["EXCLUSIVE_MINIMUM"] = "exclusiveMinimum";
    CustomFieldValidators["FORMAT"] = "format";
    CustomFieldValidators["MAX_ITEMS"] = "maxItems";
    CustomFieldValidators["MAX_LENGTH"] = "maxLength";
    CustomFieldValidators["MAX_PROPERTIES"] = "maxProperties";
    CustomFieldValidators["MAXIMUM"] = "maximum";
    CustomFieldValidators["MIN_ITEMS"] = "minItems";
    CustomFieldValidators["MIN_LENGTH"] = "minLength";
    CustomFieldValidators["MIN_PROPERTIES"] = "minProperties";
    CustomFieldValidators["MINIMUM"] = "minimum";
    CustomFieldValidators["MULTIPLE_OF"] = "multipleOf";
    CustomFieldValidators["NOT"] = "not";
    CustomFieldValidators["ONE_OF"] = "oneOf";
    CustomFieldValidators["PATTERN"] = "pattern";
    CustomFieldValidators["UNIQUE_ITEMS"] = "uniqueItems";
})(exports.CustomFieldValidators || (exports.CustomFieldValidators = {}));
exports.CustomSsoType = void 0;
(function (CustomSsoType) {
    CustomSsoType["OAUTH2"] = "oauth2";
})(exports.CustomSsoType || (exports.CustomSsoType = {}));
exports.DateTypeOptions = void 0;
(function (DateTypeOptions) {
    DateTypeOptions["DATE"] = "date";
    DateTypeOptions["DATETIME"] = "datetime";
})(exports.DateTypeOptions || (exports.DateTypeOptions = {}));
exports.DefaultSsoType = void 0;
(function (DefaultSsoType) {
    DefaultSsoType["FACEBOOK"] = "facebook";
    DefaultSsoType["GOOGLE"] = "google";
    DefaultSsoType["LINKEDIN"] = "linkedin";
})(exports.DefaultSsoType || (exports.DefaultSsoType = {}));
exports.EmailStatus = void 0;
(function (EmailStatus) {
    EmailStatus["BLOCKED"] = "BLOCKED";
    EmailStatus["BOUNCED"] = "BOUNCED";
    EmailStatus["CREATED"] = "CREATED";
    EmailStatus["DEFERRED"] = "DEFERRED";
    EmailStatus["DELIVERED"] = "DELIVERED";
    EmailStatus["DROPPED"] = "DROPPED";
    EmailStatus["PROCESSED"] = "PROCESSED";
    EmailStatus["SENT"] = "SENT";
})(exports.EmailStatus || (exports.EmailStatus = {}));
exports.EmailType = void 0;
(function (EmailType) {
    EmailType["COLLABORATOR_INVITATION"] = "CollaboratorInvitation";
    EmailType["EMAIL_VERIFICATION"] = "EmailVerification";
    EmailType["MEMBER_CONFIRMATION"] = "MemberConfirmation";
    EmailType["MEMBER_INVITATION"] = "MemberInvitation";
    EmailType["NOTIFICATION"] = "Notification";
    EmailType["RESET_PASSWORD"] = "ResetPassword";
    EmailType["UNKNOWN"] = "Unknown";
})(exports.EmailType || (exports.EmailType = {}));
exports.EventVerb = void 0;
(function (EventVerb) {
    EventVerb["ACCEPTED"] = "ACCEPTED";
    EventVerb["ADDED"] = "ADDED";
    EventVerb["BANNED"] = "BANNED";
    EventVerb["BLOCKED"] = "BLOCKED";
    EventVerb["CANCELED"] = "CANCELED";
    EventVerb["CREATED"] = "CREATED";
    EventVerb["DELETED"] = "DELETED";
    EventVerb["EXPIRED"] = "EXPIRED";
    EventVerb["FAILED"] = "FAILED";
    EventVerb["FOLLOWED"] = "FOLLOWED";
    EventVerb["HIDDEN"] = "HIDDEN";
    EventVerb["PERMITTED"] = "PERMITTED";
    EventVerb["PINGED"] = "PINGED";
    EventVerb["PINNED"] = "PINNED";
    EventVerb["PUBLISHED"] = "PUBLISHED";
    EventVerb["REJECTED"] = "REJECTED";
    EventVerb["REMOVED"] = "REMOVED";
    EventVerb["REVOKED"] = "REVOKED";
    EventVerb["UNBLOCKED"] = "UNBLOCKED";
    EventVerb["UNFOLLOWED"] = "UNFOLLOWED";
    EventVerb["UNHIDDEN"] = "UNHIDDEN";
    EventVerb["UNPINNED"] = "UNPINNED";
    EventVerb["UNPUBLISHED"] = "UNPUBLISHED";
    EventVerb["UNVERIFIED"] = "UNVERIFIED";
    EventVerb["UPDATED"] = "UPDATED";
    EventVerb["VERIFIED"] = "VERIFIED";
    EventVerb["VIEWED"] = "VIEWED";
})(exports.EventVerb || (exports.EventVerb = {}));
exports.ExtraPropertyEntityType = void 0;
(function (ExtraPropertyEntityType) {
    ExtraPropertyEntityType["MEMBER"] = "Member";
    ExtraPropertyEntityType["NETWORK"] = "Network";
})(exports.ExtraPropertyEntityType || (exports.ExtraPropertyEntityType = {}));
exports.Filtername = void 0;
(function (Filtername) {
    Filtername["AND"] = "and";
    Filtername["LEAF"] = "leaf";
    Filtername["OR"] = "or";
})(exports.Filtername || (exports.Filtername = {}));
exports.FlaggedType = void 0;
(function (FlaggedType) {
    FlaggedType["MEMBER"] = "MEMBER";
    FlaggedType["SYSTEM"] = "SYSTEM";
})(exports.FlaggedType || (exports.FlaggedType = {}));
exports.HighlightedTagType = void 0;
(function (HighlightedTagType) {
    HighlightedTagType["SECTION"] = "SECTION";
    HighlightedTagType["TOPIC"] = "TOPIC";
})(exports.HighlightedTagType || (exports.HighlightedTagType = {}));
exports.ImportRequestStage = void 0;
(function (ImportRequestStage) {
    ImportRequestStage["ANSWER"] = "ANSWER";
    ImportRequestStage["COMMENT"] = "COMMENT";
    ImportRequestStage["DISCUSSION"] = "DISCUSSION";
    ImportRequestStage["FIX_POST_TYPES"] = "FIX_POST_TYPES";
    ImportRequestStage["MEMBER"] = "MEMBER";
    ImportRequestStage["QUESTION"] = "QUESTION";
    ImportRequestStage["SPACE"] = "SPACE";
    ImportRequestStage["TOPIC"] = "TOPIC";
})(exports.ImportRequestStage || (exports.ImportRequestStage = {}));
exports.ImportRequestStatus = void 0;
(function (ImportRequestStatus) {
    ImportRequestStatus["DONE"] = "DONE";
    ImportRequestStatus["IN_PROGRESS"] = "IN_PROGRESS";
})(exports.ImportRequestStatus || (exports.ImportRequestStatus = {}));
exports.InvoiceStatus = void 0;
(function (InvoiceStatus) {
    InvoiceStatus["CANCELLED"] = "cancelled";
    InvoiceStatus["FAILED"] = "failed";
    InvoiceStatus["PAID"] = "paid";
    InvoiceStatus["PENDING"] = "pending";
})(exports.InvoiceStatus || (exports.InvoiceStatus = {}));
exports.MemberEmailStatus = void 0;
(function (MemberEmailStatus) {
    MemberEmailStatus["NOT_DELIVERED"] = "notDelivered";
    MemberEmailStatus["SENT"] = "sent";
    MemberEmailStatus["SPAMMED"] = "spammed";
    MemberEmailStatus["VERIFIED"] = "verified";
})(exports.MemberEmailStatus || (exports.MemberEmailStatus = {}));
exports.MemberInvitationStatus = void 0;
(function (MemberInvitationStatus) {
    MemberInvitationStatus["ACCEPTED"] = "accepted";
    MemberInvitationStatus["DELIVERED"] = "delivered";
    MemberInvitationStatus["NOT_DELIVERED"] = "notDelivered";
    MemberInvitationStatus["NOT_SENT"] = "notSent";
    MemberInvitationStatus["REJECTED"] = "rejected";
    MemberInvitationStatus["SENT"] = "sent";
    MemberInvitationStatus["SPAMMED"] = "spammed";
})(exports.MemberInvitationStatus || (exports.MemberInvitationStatus = {}));
exports.MemberListFilterByOperator = void 0;
(function (MemberListFilterByOperator) {
    MemberListFilterByOperator["CONTAINS"] = "contains";
    MemberListFilterByOperator["EQUALS"] = "equals";
    MemberListFilterByOperator["GT"] = "gt";
    MemberListFilterByOperator["GTE"] = "gte";
    MemberListFilterByOperator["LT"] = "lt";
    MemberListFilterByOperator["LTE"] = "lte";
    MemberListFilterByOperator["NOT"] = "not";
})(exports.MemberListFilterByOperator || (exports.MemberListFilterByOperator = {}));
exports.MemberStatus = void 0;
(function (MemberStatus) {
    MemberStatus["BLOCKED"] = "BLOCKED";
    MemberStatus["DELETED"] = "DELETED";
    MemberStatus["REMOVED"] = "REMOVED";
    MemberStatus["UNVERIFIED"] = "UNVERIFIED";
    MemberStatus["VERIFIED"] = "VERIFIED";
})(exports.MemberStatus || (exports.MemberStatus = {}));
exports.MemberStatusInput = void 0;
(function (MemberStatusInput) {
    MemberStatusInput["BLOCKED"] = "BLOCKED";
    MemberStatusInput["UNVERIFIED"] = "UNVERIFIED";
    MemberStatusInput["VERIFIED"] = "VERIFIED";
})(exports.MemberStatusInput || (exports.MemberStatusInput = {}));
exports.ModerationEntityType = void 0;
(function (ModerationEntityType) {
    ModerationEntityType["MEMBER"] = "MEMBER";
    ModerationEntityType["POST"] = "POST";
})(exports.ModerationEntityType || (exports.ModerationEntityType = {}));
exports.ModerationStatus = void 0;
(function (ModerationStatus) {
    ModerationStatus["ACCEPTED"] = "ACCEPTED";
    ModerationStatus["REJECTED"] = "REJECTED";
    ModerationStatus["REVIEW"] = "REVIEW";
})(exports.ModerationStatus || (exports.ModerationStatus = {}));
exports.NavigationItemType = void 0;
(function (NavigationItemType) {
    NavigationItemType["PRIMARY_BUTTON"] = "PRIMARY_BUTTON";
    NavigationItemType["PRIMARY_LINK"] = "PRIMARY_LINK";
    NavigationItemType["SECONDARY_BUTTON"] = "SECONDARY_BUTTON";
    NavigationItemType["TEXT_LINK"] = "TEXT_LINK";
})(exports.NavigationItemType || (exports.NavigationItemType = {}));
exports.NetworkIndustryType = void 0;
(function (NetworkIndustryType) {
    NetworkIndustryType["BLOGGER_COACH_CREATOR"] = "BloggerCoachCreator";
    NetworkIndustryType["CONSULTING_AND_AGENCY"] = "ConsultingAndAgency";
    NetworkIndustryType["ECOMERCE_AND_RETAIL"] = "EcomerceAndRetail";
    NetworkIndustryType["ENTERTAINMENT_AND_EVENTS"] = "EntertainmentAndEvents";
    NetworkIndustryType["FINANCIAN_SERVICES"] = "FinancianServices";
    NetworkIndustryType["HEALTHCARE"] = "Healthcare";
    NetworkIndustryType["INTERNET_AND_ONLINE_SERVICE"] = "InternetAndOnlineService";
    NetworkIndustryType["MEDIA_AND_PUBLISHING"] = "MediaAndPublishing";
    NetworkIndustryType["NON_PROFIT_AND_ASSOCIATION"] = "NonProfitAndAssociation";
    NetworkIndustryType["ONLINE_TRAINING_AND_EDUCATION"] = "OnlineTrainingAndEducation";
    NetworkIndustryType["OTHER"] = "Other";
    NetworkIndustryType["SOFTWARE_AND_SAAS"] = "SoftwareAndSaas";
})(exports.NetworkIndustryType || (exports.NetworkIndustryType = {}));
exports.NetworkLandingPage = void 0;
(function (NetworkLandingPage) {
    NetworkLandingPage["EXPLORE"] = "EXPLORE";
    NetworkLandingPage["FEED"] = "FEED";
    NetworkLandingPage["SPACES"] = "SPACES";
})(exports.NetworkLandingPage || (exports.NetworkLandingPage = {}));
exports.NetworkMembership = void 0;
(function (NetworkMembership) {
    NetworkMembership["INVITE_ONLY"] = "inviteOnly";
    NetworkMembership["OPEN"] = "open";
})(exports.NetworkMembership || (exports.NetworkMembership = {}));
exports.NetworkPasswordComplexity = void 0;
(function (NetworkPasswordComplexity) {
    NetworkPasswordComplexity["COMPLEX"] = "Complex";
    NetworkPasswordComplexity["SIMPLE"] = "Simple";
    NetworkPasswordComplexity["STANDARD"] = "Standard";
})(exports.NetworkPasswordComplexity || (exports.NetworkPasswordComplexity = {}));
exports.NetworkPrimaryMembersType = void 0;
(function (NetworkPrimaryMembersType) {
    NetworkPrimaryMembersType["CUSTOMERS"] = "Customers";
    NetworkPrimaryMembersType["EMPLOYEES"] = "Employees";
    NetworkPrimaryMembersType["MEMBERS"] = "Members";
    NetworkPrimaryMembersType["OTHER"] = "Other";
    NetworkPrimaryMembersType["STUDENTS"] = "Students";
})(exports.NetworkPrimaryMembersType || (exports.NetworkPrimaryMembersType = {}));
exports.NetworkReleaseChannelType = void 0;
(function (NetworkReleaseChannelType) {
    NetworkReleaseChannelType["EDGE"] = "edge";
    NetworkReleaseChannelType["STABLE"] = "stable";
})(exports.NetworkReleaseChannelType || (exports.NetworkReleaseChannelType = {}));
exports.NetworkStatus = void 0;
(function (NetworkStatus) {
    NetworkStatus["ARCHIVED"] = "archived";
    NetworkStatus["PUBLISHED"] = "published";
    NetworkStatus["UNPUBLISHED"] = "unpublished";
})(exports.NetworkStatus || (exports.NetworkStatus = {}));
exports.NetworkStatusChangedBy = void 0;
(function (NetworkStatusChangedBy) {
    NetworkStatusChangedBy["ADMIN"] = "admin";
    NetworkStatusChangedBy["SUPPORT"] = "support";
    NetworkStatusChangedBy["SYSTEM"] = "system";
})(exports.NetworkStatusChangedBy || (exports.NetworkStatusChangedBy = {}));
exports.NetworkStatusReason = void 0;
(function (NetworkStatusReason) {
    NetworkStatusReason["MEMBER_CAPACITY_EXCEEDED"] = "memberCapacityExceeded";
    NetworkStatusReason["NETWORK_CREATED"] = "networkCreated";
    NetworkStatusReason["NETWORK_LOCK_LIFTED"] = "networkLockLifted";
    NetworkStatusReason["SEATS_CAPACITY_EXCEEDED"] = "seatsCapacityExceeded";
})(exports.NetworkStatusReason || (exports.NetworkStatusReason = {}));
exports.NetworkTimeframeType = void 0;
(function (NetworkTimeframeType) {
    NetworkTimeframeType["FEW_MONTHS"] = "FewMonths";
    NetworkTimeframeType["FEW_WEEKS"] = "FewWeeks";
    NetworkTimeframeType["NOT_SURE"] = "NotSure";
    NetworkTimeframeType["VERY_SOON"] = "VerySoon";
})(exports.NetworkTimeframeType || (exports.NetworkTimeframeType = {}));
exports.NetworkVisibility = void 0;
(function (NetworkVisibility) {
    NetworkVisibility["PRIVATE"] = "private";
    NetworkVisibility["PUBLIC"] = "public";
})(exports.NetworkVisibility || (exports.NetworkVisibility = {}));
exports.NotificationChannel = void 0;
(function (NotificationChannel) {
    NotificationChannel["DESKTOP"] = "DESKTOP";
    NotificationChannel["EMAIL"] = "EMAIL";
    NotificationChannel["IN_APP"] = "IN_APP";
})(exports.NotificationChannel || (exports.NotificationChannel = {}));
exports.NotificationReason = void 0;
(function (NotificationReason) {
    NotificationReason["NETWORK_ADMIN"] = "NETWORK_ADMIN";
    NotificationReason["NETWORK_SUBSCRIPTION"] = "NETWORK_SUBSCRIPTION";
    NotificationReason["POST_AUTHOR"] = "POST_AUTHOR";
    NotificationReason["POST_SUBSCRIPTION"] = "POST_SUBSCRIPTION";
    NotificationReason["SPACE_ADMIN"] = "SPACE_ADMIN";
    NotificationReason["SPACE_SUBSCRIPTION"] = "SPACE_SUBSCRIPTION";
})(exports.NotificationReason || (exports.NotificationReason = {}));
exports.NotificationVerb = void 0;
(function (NotificationVerb) {
    NotificationVerb["COMMENT_CREATED"] = "COMMENT_CREATED";
    NotificationVerb["COMMENT_DELETED"] = "COMMENT_DELETED";
    NotificationVerb["COMMENT_UPDATED"] = "COMMENT_UPDATED";
    NotificationVerb["JOIN_REQUEST_STATUS_UPDATED"] = "JOIN_REQUEST_STATUS_UPDATED";
    NotificationVerb["MEMBER_MENTIONED"] = "MEMBER_MENTIONED";
    NotificationVerb["POST_CREATED"] = "POST_CREATED";
    NotificationVerb["POST_DELETED"] = "POST_DELETED";
    NotificationVerb["POST_FOLLOWED"] = "POST_FOLLOWED";
    NotificationVerb["POST_UPDATED"] = "POST_UPDATED";
    NotificationVerb["REACTION_CREATED"] = "REACTION_CREATED";
    NotificationVerb["REPLY_CREATED"] = "REPLY_CREATED";
    NotificationVerb["SPACE_MEMBER_ADDED"] = "SPACE_MEMBER_ADDED";
})(exports.NotificationVerb || (exports.NotificationVerb = {}));
exports.NumberTypeOptions = void 0;
(function (NumberTypeOptions) {
    NumberTypeOptions["INTEGER"] = "integer";
    NumberTypeOptions["NUMBER"] = "number";
})(exports.NumberTypeOptions || (exports.NumberTypeOptions = {}));
exports.PageSeoTagType = void 0;
(function (PageSeoTagType) {
    PageSeoTagType["LINK"] = "LINK";
    PageSeoTagType["META"] = "META";
    PageSeoTagType["TITLE"] = "TITLE";
})(exports.PageSeoTagType || (exports.PageSeoTagType = {}));
exports.PayloadType = void 0;
(function (PayloadType) {
    PayloadType["JOIN_REQUEST_STATUS"] = "JOIN_REQUEST_STATUS";
    PayloadType["MEMBER"] = "MEMBER";
    PayloadType["NETWORK"] = "NETWORK";
    PayloadType["POST"] = "POST";
    PayloadType["REACTION"] = "REACTION";
    PayloadType["SPACE"] = "SPACE";
})(exports.PayloadType || (exports.PayloadType = {}));
exports.PermissionContext = void 0;
(function (PermissionContext) {
    PermissionContext["MEMBER"] = "MEMBER";
    PermissionContext["NETWORK"] = "NETWORK";
    PermissionContext["POST"] = "POST";
    PermissionContext["SPACE"] = "SPACE";
})(exports.PermissionContext || (exports.PermissionContext = {}));
exports.PinnedInto = void 0;
(function (PinnedInto) {
    PinnedInto["MEMBER"] = "member";
    PinnedInto["NETWORK"] = "network";
    PinnedInto["POST"] = "post";
    PinnedInto["SPACE"] = "space";
})(exports.PinnedInto || (exports.PinnedInto = {}));
exports.PlanName = void 0;
(function (PlanName) {
    PlanName["BASIC"] = "basic";
    PlanName["ENTERPRISE"] = "enterprise";
    PlanName["PLUS"] = "plus";
    PlanName["PREMIUM"] = "premium";
})(exports.PlanName || (exports.PlanName = {}));
exports.PlanRenewalType = void 0;
(function (PlanRenewalType) {
    PlanRenewalType["MONTH"] = "month";
    PlanRenewalType["YEAR"] = "year";
})(exports.PlanRenewalType || (exports.PlanRenewalType = {}));
exports.PostFieldsEnum = void 0;
(function (PostFieldsEnum) {
    PostFieldsEnum["BOOLEAN1"] = "boolean1";
    PostFieldsEnum["BOOLEAN2"] = "boolean2";
    PostFieldsEnum["BOOLEAN3"] = "boolean3";
    PostFieldsEnum["BOOLEAN4"] = "boolean4";
    PostFieldsEnum["BOOLEAN5"] = "boolean5";
    PostFieldsEnum["DATE1"] = "date1";
    PostFieldsEnum["DATE2"] = "date2";
    PostFieldsEnum["DATE3"] = "date3";
    PostFieldsEnum["DATE4"] = "date4";
    PostFieldsEnum["DATE5"] = "date5";
    PostFieldsEnum["HTML1"] = "html1";
    PostFieldsEnum["HTML2"] = "html2";
    PostFieldsEnum["HTML3"] = "html3";
    PostFieldsEnum["HTML4"] = "html4";
    PostFieldsEnum["HTML5"] = "html5";
    PostFieldsEnum["HTML6"] = "html6";
    PostFieldsEnum["HTML7"] = "html7";
    PostFieldsEnum["HTML8"] = "html8";
    PostFieldsEnum["HTML9"] = "html9";
    PostFieldsEnum["HTML10"] = "html10";
    PostFieldsEnum["IMAGE1"] = "image1";
    PostFieldsEnum["IMAGE2"] = "image2";
    PostFieldsEnum["IMAGE3"] = "image3";
    PostFieldsEnum["IMAGE4"] = "image4";
    PostFieldsEnum["IMAGE5"] = "image5";
    PostFieldsEnum["NUMBER1"] = "number1";
    PostFieldsEnum["NUMBER2"] = "number2";
    PostFieldsEnum["NUMBER3"] = "number3";
    PostFieldsEnum["NUMBER4"] = "number4";
    PostFieldsEnum["NUMBER5"] = "number5";
    PostFieldsEnum["NUMBER_ARRAY1"] = "numberArray1";
    PostFieldsEnum["NUMBER_ARRAY2"] = "numberArray2";
    PostFieldsEnum["NUMBER_ARRAY3"] = "numberArray3";
    PostFieldsEnum["NUMBER_ARRAY4"] = "numberArray4";
    PostFieldsEnum["NUMBER_ARRAY5"] = "numberArray5";
    PostFieldsEnum["TEXT1"] = "text1";
    PostFieldsEnum["TEXT2"] = "text2";
    PostFieldsEnum["TEXT3"] = "text3";
    PostFieldsEnum["TEXT4"] = "text4";
    PostFieldsEnum["TEXT5"] = "text5";
    PostFieldsEnum["TEXT6"] = "text6";
    PostFieldsEnum["TEXT7"] = "text7";
    PostFieldsEnum["TEXT8"] = "text8";
    PostFieldsEnum["TEXT9"] = "text9";
    PostFieldsEnum["TEXT10"] = "text10";
    PostFieldsEnum["TEXT_ARRAY1"] = "textArray1";
    PostFieldsEnum["TEXT_ARRAY2"] = "textArray2";
    PostFieldsEnum["TEXT_ARRAY3"] = "textArray3";
    PostFieldsEnum["TEXT_ARRAY4"] = "textArray4";
    PostFieldsEnum["TEXT_ARRAY5"] = "textArray5";
})(exports.PostFieldsEnum || (exports.PostFieldsEnum = {}));
exports.PostListFilterByEnum = void 0;
(function (PostListFilterByEnum) {
    PostListFilterByEnum["BOOLEAN1"] = "boolean1";
    PostListFilterByEnum["BOOLEAN2"] = "boolean2";
    PostListFilterByEnum["BOOLEAN3"] = "boolean3";
    PostListFilterByEnum["BOOLEAN4"] = "boolean4";
    PostListFilterByEnum["BOOLEAN5"] = "boolean5";
    PostListFilterByEnum["CREATED_AT"] = "createdAt";
    PostListFilterByEnum["DATE1"] = "date1";
    PostListFilterByEnum["DATE2"] = "date2";
    PostListFilterByEnum["DATE3"] = "date3";
    PostListFilterByEnum["DATE4"] = "date4";
    PostListFilterByEnum["DATE5"] = "date5";
    PostListFilterByEnum["NUMBER1"] = "number1";
    PostListFilterByEnum["NUMBER2"] = "number2";
    PostListFilterByEnum["NUMBER3"] = "number3";
    PostListFilterByEnum["NUMBER4"] = "number4";
    PostListFilterByEnum["NUMBER5"] = "number5";
    PostListFilterByEnum["UPDATED_AT"] = "updatedAt";
})(exports.PostListFilterByEnum || (exports.PostListFilterByEnum = {}));
exports.PostListFilterByOperator = void 0;
(function (PostListFilterByOperator) {
    PostListFilterByOperator["EQUALS"] = "equals";
    PostListFilterByOperator["GT"] = "gt";
    PostListFilterByOperator["GTE"] = "gte";
    PostListFilterByOperator["IN"] = "in";
    PostListFilterByOperator["LT"] = "lt";
    PostListFilterByOperator["LTE"] = "lte";
    PostListFilterByOperator["NOT"] = "not";
})(exports.PostListFilterByOperator || (exports.PostListFilterByOperator = {}));
exports.PostListOrderByEnum = void 0;
(function (PostListOrderByEnum) {
    PostListOrderByEnum["BOOLEAN1"] = "boolean1";
    PostListOrderByEnum["BOOLEAN2"] = "boolean2";
    PostListOrderByEnum["BOOLEAN3"] = "boolean3";
    PostListOrderByEnum["BOOLEAN4"] = "boolean4";
    PostListOrderByEnum["BOOLEAN5"] = "boolean5";
    PostListOrderByEnum["CREATED_AT"] = "createdAt";
    PostListOrderByEnum["DATE1"] = "date1";
    PostListOrderByEnum["DATE2"] = "date2";
    PostListOrderByEnum["DATE3"] = "date3";
    PostListOrderByEnum["DATE4"] = "date4";
    PostListOrderByEnum["DATE5"] = "date5";
    PostListOrderByEnum["NEGATIVE_REACTIONS_COUNT"] = "negativeReactionsCount";
    PostListOrderByEnum["NUMBER1"] = "number1";
    PostListOrderByEnum["NUMBER2"] = "number2";
    PostListOrderByEnum["NUMBER3"] = "number3";
    PostListOrderByEnum["NUMBER4"] = "number4";
    PostListOrderByEnum["NUMBER5"] = "number5";
    PostListOrderByEnum["POSITIVE_REACTIONS_COUNT"] = "positiveReactionsCount";
    PostListOrderByEnum["REACTIONS_COUNT"] = "reactionsCount";
    PostListOrderByEnum["REPLIES_COUNT"] = "repliesCount";
    PostListOrderByEnum["TEXT1"] = "text1";
    PostListOrderByEnum["TEXT2"] = "text2";
    PostListOrderByEnum["TEXT3"] = "text3";
    PostListOrderByEnum["TEXT4"] = "text4";
    PostListOrderByEnum["TEXT5"] = "text5";
    PostListOrderByEnum["TOTAL_REPLIES_COUNT"] = "totalRepliesCount";
    PostListOrderByEnum["UPDATED_AT"] = "updatedAt";
})(exports.PostListOrderByEnum || (exports.PostListOrderByEnum = {}));
exports.PostMappingTypeEnum = void 0;
(function (PostMappingTypeEnum) {
    PostMappingTypeEnum["BOOLEAN"] = "boolean";
    PostMappingTypeEnum["DATE"] = "date";
    PostMappingTypeEnum["HTML"] = "html";
    PostMappingTypeEnum["IMAGE"] = "image";
    PostMappingTypeEnum["NUMBER"] = "number";
    PostMappingTypeEnum["NUMBER_ARRAY"] = "numberArray";
    PostMappingTypeEnum["TEXT"] = "text";
    PostMappingTypeEnum["TEXT_ARRAY"] = "textArray";
})(exports.PostMappingTypeEnum || (exports.PostMappingTypeEnum = {}));
exports.PostStatus = void 0;
(function (PostStatus) {
    PostStatus["ARCHIVED"] = "ARCHIVED";
    PostStatus["BLOCKED"] = "BLOCKED";
    PostStatus["DELETED"] = "DELETED";
    PostStatus["DRAFTED"] = "DRAFTED";
    PostStatus["PUBLISHED"] = "PUBLISHED";
})(exports.PostStatus || (exports.PostStatus = {}));
exports.PostTypeContext = void 0;
(function (PostTypeContext) {
    PostTypeContext["POST"] = "post";
    PostTypeContext["REPLY"] = "reply";
})(exports.PostTypeContext || (exports.PostTypeContext = {}));
exports.PrimaryScopes = void 0;
(function (PrimaryScopes) {
    PrimaryScopes["ALL_ACCESS"] = "ALL_ACCESS";
    PrimaryScopes["ASSIGN_BADGE"] = "ASSIGN_BADGE";
    PrimaryScopes["IMPERSONATE_MEMBER"] = "IMPERSONATE_MEMBER";
    PrimaryScopes["UPDATE_BADGE"] = "UPDATE_BADGE";
    PrimaryScopes["UPDATE_NETWORK"] = "UPDATE_NETWORK";
    PrimaryScopes["VIEW_APP"] = "VIEW_APP";
    PrimaryScopes["VIEW_APP_INSTALLATION"] = "VIEW_APP_INSTALLATION";
    PrimaryScopes["VIEW_BILLING_SUBSCRIPTION"] = "VIEW_BILLING_SUBSCRIPTION";
    PrimaryScopes["VIEW_IMPORT_REQUEST"] = "VIEW_IMPORT_REQUEST";
    PrimaryScopes["VIEW_MEMBER"] = "VIEW_MEMBER";
    PrimaryScopes["VIEW_MEMBER_INVITATION"] = "VIEW_MEMBER_INVITATION";
    PrimaryScopes["VIEW_MODERATION"] = "VIEW_MODERATION";
    PrimaryScopes["VIEW_NETWORK"] = "VIEW_NETWORK";
    PrimaryScopes["VIEW_NETWORK_PLAN"] = "VIEW_NETWORK_PLAN";
    PrimaryScopes["VIEW_ORGANIZATION"] = "VIEW_ORGANIZATION";
    PrimaryScopes["VIEW_POST"] = "VIEW_POST";
    PrimaryScopes["VIEW_ROLE"] = "VIEW_ROLE";
    PrimaryScopes["VIEW_SPACE"] = "VIEW_SPACE";
    PrimaryScopes["VIEW_SPACE_COLLECTION"] = "VIEW_SPACE_COLLECTION";
    PrimaryScopes["VIEW_SPACE_JOIN_REQUEST"] = "VIEW_SPACE_JOIN_REQUEST";
    PrimaryScopes["VIEW_SPACE_MEMBERSHIP"] = "VIEW_SPACE_MEMBERSHIP";
    PrimaryScopes["VIEW_SPACE_ROLE"] = "VIEW_SPACE_ROLE";
    PrimaryScopes["VIEW_SSO"] = "VIEW_SSO";
    PrimaryScopes["VIEW_SSO_MEMBERSHIP"] = "VIEW_SSO_MEMBERSHIP";
    PrimaryScopes["VIEW_SYNC_EVENT"] = "VIEW_SYNC_EVENT";
    PrimaryScopes["VIEW_TRACKER"] = "VIEW_TRACKER";
})(exports.PrimaryScopes || (exports.PrimaryScopes = {}));
exports.ReactionType = void 0;
(function (ReactionType) {
    ReactionType["EMOJI_BASE"] = "EMOJI_BASE";
    ReactionType["LIKE_BASE"] = "LIKE_BASE";
    ReactionType["VOTE_BASE"] = "VOTE_BASE";
})(exports.ReactionType || (exports.ReactionType = {}));
exports.RelationTypeOptions = void 0;
(function (RelationTypeOptions) {
    RelationTypeOptions["MEDIA"] = "media";
    RelationTypeOptions["MEMBER"] = "member";
    RelationTypeOptions["POST"] = "post";
    RelationTypeOptions["TAG"] = "tag";
})(exports.RelationTypeOptions || (exports.RelationTypeOptions = {}));
exports.ReportCategory = void 0;
(function (ReportCategory) {
    ReportCategory["HARASSMENT"] = "HARASSMENT";
    ReportCategory["MISINFORMATION"] = "MISINFORMATION";
    ReportCategory["NUDITY"] = "NUDITY";
    ReportCategory["SPAM"] = "SPAM";
    ReportCategory["SUICIDE"] = "SUICIDE";
    ReportCategory["TERRORISM"] = "TERRORISM";
    ReportCategory["VIOLENCE"] = "VIOLENCE";
})(exports.ReportCategory || (exports.ReportCategory = {}));
exports.ReportDataType = void 0;
(function (ReportDataType) {
    ReportDataType["CHART_DATA"] = "chartData";
    ReportDataType["ENTITY_REPORT"] = "entityReport";
    ReportDataType["INT_VALUE"] = "intValue";
    ReportDataType["STRING_VALUE"] = "stringValue";
})(exports.ReportDataType || (exports.ReportDataType = {}));
exports.ReportSlug = void 0;
(function (ReportSlug) {
    ReportSlug["ACTIVE_MEMBERS"] = "activeMembers";
    ReportSlug["AVERAGE_DAILY_ACTIVE_MEMBERS"] = "averageDailyActiveMembers";
    ReportSlug["HIGHLIGHTS"] = "highlights";
    ReportSlug["MEMBER_SCORE"] = "memberScore";
    ReportSlug["NEW_MEMBERS_OVER_TIME"] = "newMembersOverTime";
    ReportSlug["NEW_POSTS"] = "newPosts";
    ReportSlug["NEW_REACTIONS"] = "newReactions";
    ReportSlug["NEW_REPLIES"] = "newReplies";
    ReportSlug["POPULAR_DAYS_OF_WEEK"] = "popularDaysOfWeek";
    ReportSlug["POPULAR_HOURS_OF_DAY"] = "popularHoursOfDay";
    ReportSlug["POSTS_V_SREPLIES"] = "postsVSreplies";
    ReportSlug["TOP_MEMBERS"] = "topMembers";
    ReportSlug["TOP_POSTS"] = "topPosts";
    ReportSlug["TOP_SPACES"] = "topSpaces";
    ReportSlug["TOTAL_MEMBERS"] = "totalMembers";
    ReportSlug["TOTAL_POSTS"] = "totalPosts";
    ReportSlug["TOTAL_REACTIONS"] = "totalReactions";
    ReportSlug["TOTAL_REPLIES"] = "totalReplies";
    ReportSlug["TOTAL_VISITORS"] = "totalVisitors";
    ReportSlug["TRENDING_TAGS"] = "trendingTags";
})(exports.ReportSlug || (exports.ReportSlug = {}));
exports.ReportTimeFrame = void 0;
(function (ReportTimeFrame) {
    ReportTimeFrame["ALL_TIME"] = "allTime";
    ReportTimeFrame["LAST_CALENDAR_QUARTER"] = "lastCalendarQuarter";
    ReportTimeFrame["LAST_CALENDAR_YEAR"] = "lastCalendarYear";
    ReportTimeFrame["LAST_MONTH"] = "lastMonth";
    ReportTimeFrame["LAST_NINETY_DAYS"] = "lastNinetyDays";
    ReportTimeFrame["LAST_SEVEN_DAYS"] = "lastSevenDays";
    ReportTimeFrame["LAST_THIRTY_DAYS"] = "lastThirtyDays";
    ReportTimeFrame["LAST_TWELVE_MONTH"] = "lastTwelveMonth";
    ReportTimeFrame["LAST_WEEK"] = "lastWeek";
    ReportTimeFrame["TODAY"] = "today";
    ReportTimeFrame["YESTERDAY"] = "yesterday";
})(exports.ReportTimeFrame || (exports.ReportTimeFrame = {}));
exports.ReportableEntityType = void 0;
(function (ReportableEntityType) {
    ReportableEntityType["MEMBER"] = "member";
    ReportableEntityType["POST"] = "post";
    ReportableEntityType["SPACE"] = "space";
    ReportableEntityType["TOPIC"] = "topic";
})(exports.ReportableEntityType || (exports.ReportableEntityType = {}));
exports.RichTextTypeOptions = void 0;
(function (RichTextTypeOptions) {
    RichTextTypeOptions["HTML"] = "html";
    RichTextTypeOptions["MARKUP"] = "markup";
})(exports.RichTextTypeOptions || (exports.RichTextTypeOptions = {}));
exports.RoleListOrderByEnum = void 0;
(function (RoleListOrderByEnum) {
    RoleListOrderByEnum["CREATED_AT"] = "CREATED_AT";
    RoleListOrderByEnum["UPDATED_AT"] = "UPDATED_AT";
})(exports.RoleListOrderByEnum || (exports.RoleListOrderByEnum = {}));
exports.RoleType = void 0;
(function (RoleType) {
    RoleType["ADMIN"] = "admin";
    RoleType["GUEST"] = "guest";
    RoleType["MEMBER"] = "member";
    RoleType["MODERATOR"] = "moderator";
})(exports.RoleType || (exports.RoleType = {}));
exports.SearchEntityType = void 0;
(function (SearchEntityType) {
    SearchEntityType["MEMBER"] = "member";
    SearchEntityType["POST"] = "post";
    SearchEntityType["SPACE"] = "space";
})(exports.SearchEntityType || (exports.SearchEntityType = {}));
exports.SortOrder = void 0;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(exports.SortOrder || (exports.SortOrder = {}));
exports.SpaceJoinRequestStatus = void 0;
(function (SpaceJoinRequestStatus) {
    SpaceJoinRequestStatus["COMPLETED"] = "COMPLETED";
    SpaceJoinRequestStatus["DECLINED"] = "DECLINED";
    SpaceJoinRequestStatus["PENDING"] = "PENDING";
})(exports.SpaceJoinRequestStatus || (exports.SpaceJoinRequestStatus = {}));
exports.SpaceListOrderByEnum = void 0;
(function (SpaceListOrderByEnum) {
    SpaceListOrderByEnum["CREATED_AT"] = "CREATED_AT";
    SpaceListOrderByEnum["CUSTOM_ORDERING_INDEX"] = "CUSTOM_ORDERING_INDEX";
    SpaceListOrderByEnum["UPDATED_AT"] = "UPDATED_AT";
})(exports.SpaceListOrderByEnum || (exports.SpaceListOrderByEnum = {}));
exports.SpaceMemberListOrderByEnum = void 0;
(function (SpaceMemberListOrderByEnum) {
    SpaceMemberListOrderByEnum["CREATED_AT"] = "CREATED_AT";
    SpaceMemberListOrderByEnum["UPDATED_AT"] = "UPDATED_AT";
})(exports.SpaceMemberListOrderByEnum || (exports.SpaceMemberListOrderByEnum = {}));
exports.SpaceMembershipStatus = void 0;
(function (SpaceMembershipStatus) {
    SpaceMembershipStatus["JOINED"] = "joined";
    SpaceMembershipStatus["NOT_JOINED"] = "notJoined";
    SpaceMembershipStatus["REQUESTED"] = "requested";
})(exports.SpaceMembershipStatus || (exports.SpaceMembershipStatus = {}));
exports.SpaceNotificationPreference = void 0;
(function (SpaceNotificationPreference) {
    SpaceNotificationPreference["ALL"] = "ALL";
    SpaceNotificationPreference["NEW_POST"] = "NEW_POST";
    SpaceNotificationPreference["NONE"] = "NONE";
})(exports.SpaceNotificationPreference || (exports.SpaceNotificationPreference = {}));
exports.SpaceRoleListOrderByEnum = void 0;
(function (SpaceRoleListOrderByEnum) {
    SpaceRoleListOrderByEnum["CREATED_AT"] = "CREATED_AT";
    SpaceRoleListOrderByEnum["UPDATED_AT"] = "UPDATED_AT";
})(exports.SpaceRoleListOrderByEnum || (exports.SpaceRoleListOrderByEnum = {}));
exports.SpaceRoleType = void 0;
(function (SpaceRoleType) {
    SpaceRoleType["ADMIN"] = "admin";
    SpaceRoleType["MEMBER"] = "member";
})(exports.SpaceRoleType || (exports.SpaceRoleType = {}));
exports.SpaceType = void 0;
(function (SpaceType) {
    SpaceType["BROADCAST"] = "Broadcast";
    SpaceType["ENTITY"] = "Entity";
    SpaceType["GROUP"] = "Group";
    SpaceType["PAGE"] = "Page";
    SpaceType["PRIVATE_MESSAGE"] = "PrivateMessage";
})(exports.SpaceType || (exports.SpaceType = {}));
exports.SsoProvider = void 0;
(function (SsoProvider) {
    SsoProvider["AUTH0"] = "auth0";
    SsoProvider["CUSTOM"] = "custom";
    SsoProvider["MEMBERFUL"] = "memberful";
    SsoProvider["OKTA"] = "okta";
    SsoProvider["OUTSETA"] = "outseta";
    SsoProvider["WORDPRESS"] = "wordpress";
})(exports.SsoProvider || (exports.SsoProvider = {}));
exports.SsoStatus = void 0;
(function (SsoStatus) {
    SsoStatus["DISABLE"] = "disable";
    SsoStatus["ENABLE"] = "enable";
})(exports.SsoStatus || (exports.SsoStatus = {}));
exports.SsoType = void 0;
(function (SsoType) {
    SsoType["FACEBOOK"] = "facebook";
    SsoType["GOOGLE"] = "google";
    SsoType["JWT"] = "jwt";
    SsoType["LINKEDIN"] = "linkedin";
    SsoType["OAUTH2"] = "oauth2";
})(exports.SsoType || (exports.SsoType = {}));
exports.StoreItemStanding = void 0;
(function (StoreItemStanding) {
    StoreItemStanding["OFFICIAL"] = "OFFICIAL";
    StoreItemStanding["REGULAR"] = "REGULAR";
    StoreItemStanding["VERIFIED"] = "VERIFIED";
})(exports.StoreItemStanding || (exports.StoreItemStanding = {}));
exports.StoreItemStatus = void 0;
(function (StoreItemStatus) {
    StoreItemStatus["DELETED"] = "DELETED";
    StoreItemStatus["PRIVATE"] = "PRIVATE";
    StoreItemStatus["PUBLIC"] = "PUBLIC";
})(exports.StoreItemStatus || (exports.StoreItemStatus = {}));
exports.SubscriptionStatus = void 0;
(function (SubscriptionStatus) {
    SubscriptionStatus["ACTIVE"] = "active";
    SubscriptionStatus["CANCELED"] = "canceled";
    SubscriptionStatus["FAILED"] = "failed";
    SubscriptionStatus["INCOMPLETE"] = "incomplete";
    SubscriptionStatus["INCOMPLETE_EXPIRED"] = "incomplete_expired";
    SubscriptionStatus["PAST_DUE"] = "past_due";
    SubscriptionStatus["TRIALING"] = "trialing";
    SubscriptionStatus["UNPAID"] = "unpaid";
})(exports.SubscriptionStatus || (exports.SubscriptionStatus = {}));
exports.TagFilterType = void 0;
(function (TagFilterType) {
    TagFilterType["BLACKLIST"] = "BLACKLIST";
    TagFilterType["WHITELIST"] = "WHITELIST";
})(exports.TagFilterType || (exports.TagFilterType = {}));
exports.TagListOrderByEnum = void 0;
(function (TagListOrderByEnum) {
    TagListOrderByEnum["CREATED_AT"] = "CREATED_AT";
    TagListOrderByEnum["UPDATED_AT"] = "UPDATED_AT";
})(exports.TagListOrderByEnum || (exports.TagListOrderByEnum = {}));
exports.TemplateEntityType = void 0;
(function (TemplateEntityType) {
    TemplateEntityType["POST"] = "Post";
    TemplateEntityType["POST_TYPE"] = "PostType";
    TemplateEntityType["SPACE"] = "Space";
    TemplateEntityType["SPACE_POST_TYPE"] = "SpacePostType";
    TemplateEntityType["TAG"] = "Tag";
})(exports.TemplateEntityType || (exports.TemplateEntityType = {}));
exports.TextTypeOptions = void 0;
(function (TextTypeOptions) {
    TextTypeOptions["FULL_TEXT"] = "fullText";
    TextTypeOptions["SHORT_TEXT"] = "shortText";
})(exports.TextTypeOptions || (exports.TextTypeOptions = {}));
exports.ThemeStatus = void 0;
(function (ThemeStatus) {
    ThemeStatus["DRAFT"] = "draft";
    ThemeStatus["PUBLISHED"] = "published";
})(exports.ThemeStatus || (exports.ThemeStatus = {}));
exports.TopNavigationAlignment = void 0;
(function (TopNavigationAlignment) {
    TopNavigationAlignment["CENTER"] = "CENTER";
    TopNavigationAlignment["LEFT"] = "LEFT";
    TopNavigationAlignment["RIGHT"] = "RIGHT";
})(exports.TopNavigationAlignment || (exports.TopNavigationAlignment = {}));
exports.UnauthorizedReason = void 0;
(function (UnauthorizedReason) {
    UnauthorizedReason["ACCESS"] = "ACCESS";
    UnauthorizedReason["PLAN"] = "PLAN";
})(exports.UnauthorizedReason || (exports.UnauthorizedReason = {}));
exports.UnsubscribeTokenContext = void 0;
(function (UnsubscribeTokenContext) {
    UnsubscribeTokenContext["ALL"] = "ALL";
    UnsubscribeTokenContext["MEMBER"] = "MEMBER";
    UnsubscribeTokenContext["MENTIONS"] = "MENTIONS";
    UnsubscribeTokenContext["POST"] = "POST";
    UnsubscribeTokenContext["REACTIONS"] = "REACTIONS";
    UnsubscribeTokenContext["SPACE"] = "SPACE";
})(exports.UnsubscribeTokenContext || (exports.UnsubscribeTokenContext = {}));
exports.UseTemplateRequestStatus = void 0;
(function (UseTemplateRequestStatus) {
    UseTemplateRequestStatus["FAILED"] = "Failed";
    UseTemplateRequestStatus["IN_PROGRESS"] = "InProgress";
    UseTemplateRequestStatus["SUCCEEDED"] = "Succeeded";
})(exports.UseTemplateRequestStatus || (exports.UseTemplateRequestStatus = {}));
exports.VatType = void 0;
(function (VatType) {
    VatType["AE_TRN"] = "ae_trn";
    VatType["AU_ABN"] = "au_abn";
    VatType["AU_ARN"] = "au_arn";
    VatType["BR_CNPJ"] = "br_cnpj";
    VatType["BR_CPF"] = "br_cpf";
    VatType["CA_BN"] = "ca_bn";
    VatType["CA_GST_HST"] = "ca_gst_hst";
    VatType["CA_PST_BC"] = "ca_pst_bc";
    VatType["CA_PST_MB"] = "ca_pst_mb";
    VatType["CA_PST_SK"] = "ca_pst_sk";
    VatType["CA_QST"] = "ca_qst";
    VatType["CH_VAT"] = "ch_vat";
    VatType["CL_TIN"] = "cl_tin";
    VatType["ES_CIF"] = "es_cif";
    VatType["EU_VAT"] = "eu_vat";
    VatType["GB_VAT"] = "gb_vat";
    VatType["HK_BR"] = "hk_br";
    VatType["ID_NPWP"] = "id_npwp";
    VatType["IL_VAT"] = "il_vat";
    VatType["IN_GST"] = "in_gst";
    VatType["JP_CN"] = "jp_cn";
    VatType["JP_RN"] = "jp_rn";
    VatType["KR_BRN"] = "kr_brn";
    VatType["LI_UID"] = "li_uid";
    VatType["MX_RFC"] = "mx_rfc";
    VatType["MY_FRP"] = "my_frp";
    VatType["MY_ITN"] = "my_itn";
    VatType["MY_SST"] = "my_sst";
    VatType["NO_VAT"] = "no_vat";
    VatType["NZ_GST"] = "nz_gst";
    VatType["RU_INN"] = "ru_inn";
    VatType["RU_KPP"] = "ru_kpp";
    VatType["SA_VAT"] = "sa_vat";
    VatType["SG_GST"] = "sg_gst";
    VatType["SG_UEN"] = "sg_uen";
    VatType["TH_VAT"] = "th_vat";
    VatType["TW_VAT"] = "tw_vat";
    VatType["US_EIN"] = "us_ein";
    VatType["ZA_VAT"] = "za_vat";
})(exports.VatType || (exports.VatType = {}));
exports.WidgetContexts = void 0;
(function (WidgetContexts) {
    WidgetContexts["MEMBER"] = "MEMBER";
    WidgetContexts["POST"] = "POST";
    WidgetContexts["SPACE"] = "SPACE";
})(exports.WidgetContexts || (exports.WidgetContexts = {}));
exports.WidgetPositions = void 0;
(function (WidgetPositions) {
    WidgetPositions["PAGE"] = "PAGE";
    WidgetPositions["SIDEBAR"] = "SIDEBAR";
})(exports.WidgetPositions || (exports.WidgetPositions = {}));
